### ISSUE TITLE
lack of bidirectionality in type checking products

### ISSUE THREAD

**Issue by gares** [LPCIC/coq-elpi#465]
```coq
Check (A -> _) : Prop. (* error, Coq commits _ to be Type too early *)
Check (A -> (_ : Prop)) : Prop. (* OK *)
```

Hopefully this will be solved by our friends in Nantes (CC @ppedrot @SkySkimmer) with sort polymorphism.

**Comment by ppedrot** [comment]
This works for me on master already?

**Comment by ppedrot** [comment]
It's not a relevance problem but a sort one. Also, in master, creating a fresh sort generates a sort variable, so the old code should transparently handle this kind of situation without explicit porting.

**Comment by gares** [comment]
fixed in Coq 8.18, hacked in 4dc342a33d751f3e9d9fc2082be5e0eb6778e556
